### ISSUE TITLE
[Grid - Basics] Tooltip in header is not keyboard accessible and is not announced by screen readers

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3185]
## Current behavior

In [Grid - Basics Header story](https://storybook.sanomalearning.design/?path=/story/grid-basics--header) mouse user can expand tooltip by hovering on icon inside header.
Sighted keyboard users and screen reader users cannot set focus on icon to display tooltip. Tooltip's content is not announced by screen readers.

<img width="744" height="150" alt="Image" src="https://github.com/user-attachments/assets/a4330c33-ee8f-47bf-84db-0a63a7acb6a4" />

## Suggested solution

Changing Tooltip to Infotip should fix this issue.
This is a scenario to take into account when making https://github.com/sl-design-system/components/issues/2111

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
